pkg/util/deployment/deployment.go

```diff
@@ -48,6 +48,39 @@ const (
 	RollbackDone              = "DeploymentRollback"
 )
 
+// GetAllReplicaSets list all RSes of a given deployment atomically
+func GetAllReplicaSets(deployment *extensions.Deployment, c clientset.Interface) (allOldRSes []*extensions.ReplicaSet, newRS *extensions.ReplicaSet, err error) {
```

Comment:
Why can't we re-use this helper for all list operations in the deployment controller? Do we guarantee that the rs will have as labels the deployment selector? Related issue: https://github.com/kubernetes/kubernetes/issues/17491